Source/main.cpp

```diff
@@ -264,6 +277,49 @@ void main_main ()
             ComputeHbias(H_biasfield, time, geom);
         }
 
+        if ((Hbias_sweep == 1) && (increment_Hbias == 1)) {
+	   // We save the value of the initial Hbias so we know when to reverse the field
+           /*
+	   if (hysteresis_iter == 1){
+               
+	   }
+           */
+
+	   for (MFIter mfi(H_biasfield[0]); mfi.isValid(); ++mfi)
+           {
+               const Box& bx = mfi.tilebox();
+
+	       // extract dx from the geometry object
+	       GpuArray<Real,AMREX_SPACEDIM> dx = geom.CellSizeArray();
+
+	       // extract field data
+	       const Array4<Real>& Hx_bias = H_biasfield[0].array(mfi);
+	       const Array4<Real>& Hy_bias = H_biasfield[1].array(mfi);
+	       const Array4<Real>& Hz_bias = H_biasfield[2].array(mfi);
+
+	       amrex::ParallelFor( bx, [=] AMREX_GPU_DEVICE (int i, int j, int k)
+	       {
+	           Hx_bias(i,j,k) += sign*increment_size;
+	           Hy_bias(i,j,k) += sign*increment_size;
+	           Hz_bias(i,j,k) += sign*increment_size;
+
+
+		});
+	    }
+
+            normalized_Mx = SumNormalizedM(Ms,Mfield[0]);
+            normalized_My = SumNormalizedM(Ms,Mfield[1]);
+            normalized_Mz = SumNormalizedM(Ms,Mfield[2]);
+	    
+	    outputFile << "time = " << time << " "
+                    << "Hbias_increment: "
+                    << normalized_Mx/num_mag << " "
+                    << normalized_My/num_mag << " "
+                    << normalized_Mz/num_mag << std::endl;
+
+	    increment_Hbias = 0;
```

Comment:
@jclepelch What you want to do now is at the top of main.cpp create a global "int switch_counter=0;".  Then, here add "counter++" and also "if (counter%nsteps_hysterisis == 0) {sign *= -1;}" to flip the direction of the sweep.  nsteps_hysterisis still needs to be added to the namelist.